docs/rfcs/rfc-034-not-operator.md

```diff
@@ -0,0 +1,71 @@
+# Not operator
+
+## Problem
+
+Currently, boolean operators like `AND` and `OR` are supported by the library for filters and authorizations purpose.
+Almost all filters, for instance:
+`_CONTAINS`, `_IN`, `_STARTS_WITH` have their counterpart as:
+`_NOT_CONTAINS`, `_NOT_IN`, `_NOT_STARTS_WITH`.
+The goal of this RFC is to propose a mechanism to remove these extra filters in favor of a general NOT operator as it's already the case for the `AND` and `OR`.
+With the new operator is possible to reduce the schema size and have a more consistent API, currently, only a portion of predicates have the negate counterpart.
+
+## Proposed Solution
+
+This RFC proposes to deprecate all the previous filters with the `@deprecated` directive and deprecate these in the `4.x` release.
+A new operator `NOT` should then be included where the other two operators `AND` and `OR` are currently present.
```

Comment:
Are there any limitations/risks (in the code for instance) we have to keep in mind when using combinations of these three operators (NOT, AND, OR)?

Reply:
Not really, currently if we encounter an `AND` or a `OR` we wrap the following predicate using Cypher Builder: `Cypyher.and`/`Cypher.or`, to support the new operator we have just to wrap the following predicates inside a `Cypher.not`. It's already possible to chain predicates using CypherBuilder!

Reply:
Perfect!